File: example/python/how_to_init_client.py
"""
Examples of initializing the Catalog API client.

error 'invalid character '<' looking for beginning of value' may because of wrong endpoint or wrong proxy configured
endpoint should be like 'catalogapi.ap-southeast-5.maxcompute.aliyun.com'
"""
from maxcompute_tea_openapi import models as open_api_models
from pyodps_catalog.client import Client


def way_1():
    """Initialize with catalog endpoint directly."""
    endpoint = ""
    access_key_id = ""
    access_key_secret = ""

    config = open_api_models.Config(
        endpoint=endpoint,
        access_key_id=access_key_id,
        access_key_secret=access_key_secret,
    )
    client = Client(config)
    print(client._endpoint)


def way_2():
    """
    Initialize with ODPS endpoint only.
    The catalog API endpoint is resolved lazily on the first request via the routing API.
    odps_endpoint should be like 'service.cn-shanghai.maxcompute.aliyun.com'
    """
    odps_endpoint = ""
    access_key_id = ""
    access_key_secret = ""

    config = open_api_models.Config(
        odps_endpoint=odps_endpoint,
        protocol="http",
        access_key_id=access_key_id,
        access_key_secret=access_key_secret,
    )
    client = Client(config)
    # Endpoint is empty until the first API call triggers routing resolution.
    print(client._endpoint)


def way_3():
    """Initialize with STS token for temporary credentials."""
    endpoint = ""
    access_key_id = ""
    access_key_secret = ""
    security_token = ""

    config = open_api_models.Config(
        endpoint=endpoint,
        access_key_id=access_key_id,
        access_key_secret=access_key_secret,
        security_token=security_token,
    )
    client = Client(config)
    print(client._endpoint)

File: example/java/src/main/java/HowToInitClient.java
import com.aliyun.odps.catalog.Client;
import com.aliyun.odps.models.Config;

/**
 * Examples of initializing the Catalog API client.
 *
 * error 'invalid character '<' looking for beginning of value' may because of wrong endpoint or wrong proxy configured
 * endpoint should be like 'catalogapi.ap-southeast-5.maxcompute.aliyun.com'
 */
public class HowToInitClient {

    /**
     * Initialize with catalog endpoint directly.
     */
    static void way1() throws Exception {
        String endpoint = "";
        String accessKeyId = "";
        String accessKeySecret = "";

        Config config = new Config();
        config.setEndpoint(endpoint);
        config.setAccessKeyId(accessKeyId);
        config.setAccessKeySecret(accessKeySecret);

        Client client = new Client(config);
        System.out.println(client._endpoint);
    }

    /**
     * Initialize with ODPS endpoint only.
     * The catalog API endpoint is resolved lazily on the first request via the routing API.
     * odpsEndpoint should be like 'service.cn-shanghai.maxcompute.aliyun.com'
     */
    static void way2() throws Exception {
        String odpsEndpoint = "";
        String accessKeyId = "";
        String accessKeySecret = "";

        Config config = new Config();
        config.setOdpsEndpoint(odpsEndpoint);
        config.setProtocol("http");
        config.setAccessKeyId(accessKeyId);
        config.setAccessKeySecret(accessKeySecret);

        Client client = new Client(config);
        // Endpoint is empty until the first API call triggers routing resolution.
        System.out.println(client._endpoint);
    }

    /**
     * Initialize with STS token for temporary credentials.
     */
    static void way3() throws Exception {
        String endpoint = "";
        String accessKeyId = "";
        String accessKeySecret = "";
        String securityToken = "";

        Config config = new Config();
        config.setEndpoint(endpoint);
        config.setAccessKeyId(accessKeyId);
        config.setAccessKeySecret(accessKeySecret);
        config.setSecurityToken(securityToken);

        Client client = new Client(config);
        System.out.println(client._endpoint);
    }
}
